src/Sinch/SinchClient.cs

```diff
@@ -325,22 +326,25 @@ private SmsClient InitSms(SinchOptions optionsObj, IHttp httpSnakeCase)
 
             _logger?.LogInformation("Initializing SMS client with {project_id} in {region}", _projectId,
                 optionsObj.SmsHostingRegion.Value);
-            return new SmsClient(new ProjectId(_projectId),
+
+            return new SmsClient(
+                new ProjectId(
+                    _projectId!), // exception is throw when trying to get SMS client property if _projectId is null
```

Comment:
Null forgiving operator (!) is not about to be sure something is not null, but to suppress warnings which is necessary for the case - same applies to the OAuth. Null refs is just another tools analysis and not a panacea to fix null ref exceptions, unfortunately

Reply:
The goal of the `!` is to override the compiler analysis so that it would not issue a warning. 
If I look at the `SmsClient` constructor, I see that `ProjectId` is a record which expects a non-null value. As there is no check here that prevents the `_projectId` variable to be null, maybe it would be clearer to set the  record parameter as nullable such as `string? Value` and handle the null-value case in the `SmsClient` constructor before initializing the API classes (Batches, Inbounds...)

Reply:
This is by design, specific clients needs no nulls to function properly, but the caller is responsible  for that and manages delayed/null initialization